interfaces/tests/file_int/file_int.c

```diff
@@ -19,11 +19,13 @@
 static TEST_MUTEX_HANDLE g_testByTest;
 
 #include "azure_c_pal/sync.h"
-#include "azure_c_pal/file.h"
+#include "azure_c_pal/gballoc_hl.h"

```

Comment:
>gballoc_hl [](start = 22, length = 10)

Is gballoc_hl_redirect.h needed?

Reply:
Actually not. The reason why you'd want to use gballoc_redirect.h is to have gballoc_hl_malloc renamed to malloc. And that only is needed in the context of mocking.

But these are "int" tests so no mocking is needed. 

---
In reply to: [462448252](https://github.com/Azure/azure-c-pal/pull/21#discussion_r462448252) [](ancestors = 462448252)

Reply:
But how does that work then? gballoc_hl.h doesn't replace malloc et. al. with gballoc_hl_malloc...

---
In reply to: [463130940](https://github.com/Azure/azure-c-pal/pull/21#discussion_r463130940) [](ancestors = 463130940,462448252)

Reply:
Oh so int tests just use malloc, but the code being tested uses gballoc because it includes the redirect? And we just include gballoc_hl.h here to init from the test code?

---
In reply to: [463133981](https://github.com/Azure/azure-c-pal/pull/21#discussion_r463133981) [](ancestors = 463133981,463130940,462448252)